charms/istio-pilot/src/charm.py

```diff
@@ -179,25 +209,57 @@ def handle_ingress_auth(self, event):
             for r in auth_routes
         )
 
-        manifests = [rbac_configs, auth_filters]
+        manifests = [auth_filters]
         manifests = '\n'.join([m for m in manifests if m])
-        subprocess.run(
+        self._kubectl(
+            'delete',
+            'envoyfilters',
+            f"-lapp.juju.is/created-by={self.app.name},app.istio-pilot.io/is-workload-entity=true",
+        )
+
+        self._kubectl("apply", "-f-", input=manifests)
+
+    def _kubectl(self, *args, namespace=None, input=None, capture_output=False):
+        """Helper for running kubectl."""
+        if isinstance(input, str):
+            input = input.encode("utf-8")
+        res = subprocess.run(
             [
                 './kubectl',
-                'delete',
-                'envoyfilters,rbacconfigs',
-                f'-lapp.juju.is/created-by={self.model.app.name}',
                 '-n',
-                self.model.name,
+                namespace or self.model.name,
+                *args,
             ],
+            input=input,
+            capture_output=capture_output,
             check=True,
         )
-
-        subprocess.run(
-            ["./kubectl", "apply", "-f-"],
-            input=manifests.encode('utf-8'),
-            check=True,
+        if capture_output:
+            return res.stdout.decode("utf-8")
+
+    def _get_gateway_address(self):
+        """Look up the load balancer address for the ingress gateway.
+
+        If the gateway isn't available or doesn't have a load balancer address yet,
+        returns None.
+        """
+        svcs = yaml.safe_load(
+            self._kubectl(
+                "get",
+                "svc",
+                "-l",
+                "istio=ingressgateway",
```

Comment:
Is this a typo? Should this be `istio-ingressgateway`?

Reply:
I believe this is correct.  I think it is trying to select objects created by the istio-gateway charm in the [manifest](https://github.com/canonical/istio-operators/blob/ca-scribner/fix-ingress-auth-main/charms/istio-gateway/src/manifest.yaml), which have an `istio={{kind}}gateway` label